src/main/java/com/auction/controller/AuctionController.java

```diff
@@ -26,12 +26,17 @@ public List<Auction> getALlAuctions() {
     }
 
     @GetMapping("/{auctionId}")
-    public Auction getAuctionById(@PathVariable int auctionId) {
-        return auctionService.findById(auctionId);
+    public ResponseEntity<Auction> getAuctionById(@PathVariable int auctionId) {
+        Auction auction = auctionService.findById(auctionId);
+        if (auction != null) {
+            return ResponseEntity.ok(auction);
+        } else {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
+        }
     }
 
-    @PostMapping("/addNewAuction/{productId}")
-    public Auction addNewAuction(@RequestBody Auction theAuction, @PathVariable int productId) {
+    @PostMapping("/{productId}")
+    public Auction createAuction(@RequestBody Auction theAuction, @PathVariable int productId) {
```

Comment:
you should not use whole Object of Auction as a part of payload. If the Auction is already exists so you must just use its ID, then fetch it from DB, check if such product has not been added into it already, and only after you can update theAuctions. If it does not exist - use product Id as a part of theAuction payload